app/controllers/authenticated_controller.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# Provides authentication to derived controllers.
+# User is exposed via current_user method.
+class AuthenticatedController < ApplicationController
+  def current_user
+    @current_user ||= Authentication::UserById.new(
+      Authentication::IdFromJwt.new(
+        Http::BearerTokenFromHeaders.new(
+          request.headers
+        )
+      )
+    ).to_model
```

Comment:
Looks a little bit confusing. Wouldn't it be better to have id and token as attributes?

Reply:
I don't get it. Do you mean attributes of a controller?

Reply:
I mean use `attr_accessor`